src/Revision.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq;
+
+use Rinq\Exception\FrozenAttributeException;
+use Rinq\Exception\StaleFetchException;
+use Rinq\Exception\StaleUpdateException;
+use Rinq\Ident\Reference;
+
+/**
+ * Revision represents a specific revision of session.
+ *
+ * Revision is the sole interface for manipulating a session's attribute table.
+ *
+ * The underlying session may be "local", i.e. owned by a peer running in this
+ * process, or "remote", owned by a different peer.
+ *
+ * For remote sessions, operations may require network IO. Deadlines are
+ * honoured for all methods that accept a context.
+ */
+interface Revision
+{
+    /**
+     * The session reference, which holds the session ID and the revision number
+     * represented by this instance.
+     *
+     * @return Reference The session reference.
+     */
+    public function reference(): Reference;
+
+    /**
+     * Refresh returns the latest revision of the session.
+     *
+     * @return Revision Latest revision of the session.
+     *
+     * @throws NotFoundException If the session has been closed and revision is invalid.
+     */
+    public function refresh(Context $context): Revision;
+
+    /**
+     * Get returns the attribute with key $key from the attribute table.
+     *
+     * The returned attribute is guaranteed to be correct as of
+     * Reference::revision().
+     *
+     * Non-existent attributes are equivalent to empty attributes, therefore it
+     * is not an error to request a key that has never been created.
+     *
+     * Peers do not always have a copy of the complete attribute table. If the
+     * attribute value is unknown it is fetched from the owning peer.
+     *
+     * To fetch the attribute value at the later revision, first call refresh()
+     * then retry the get() on the newer revision.
```

Comment:
There's some info missing from this paragraph, probably because it was a bit go-specific - it looks to be correct in `getMany()` (the part about a `StaleFetchException` being thrown).